Sync usb.c and add timeout for everdrive busy wait

anacierdem opened the pull request:
Also opened [this PR](https://github.com/buu342/N64-UNFLoader/pull/65) for UNFLoader.

rasky reviewed (approved):
OK, but I'd wait for upstream to review and approve first.

anacierdem committed:
Sync usb.c and add timeout for everdrive busy wait

- Also prevent usb.c from altering PI.

anacierdem commented:
I re-synced but there is the upstream alignment change now. Should we wait for a proper fix or go ahead? We might de-vendor that file and fix as we need as well (possibly also removing libultra parts)

anacierdem committed:
Fix root Makefile and `build.sh`

- The first problem was that the root `install` recipe was only
depending on `libdragon.a` and `libdragonsys.a`, not the complete chain.
This is fixed by making install depend on `libdragon` instead.
- Second problem; example clean recipes using the include are unable to
find it when the parent makefile (`clobber` recipe) is executed. The
root Makefile has access to the build system but children does not until
it is installed. This was also causing `build.sh` to fail.
- I have also changed the sub make commands into `$(MAKE)` such that
make can know they are actually make commands. Also see [this](https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable)

rasky commented on Makefile:
```diff
@@ -87,7 +90,7 @@ install: libdragon.a libdragonsys.a
 	install -Cv -m 0644 include/ucode.S $(INSTALLDIR)/mips64-elf/include/ucode.S
 
-clean:
+clean: install-mk
```
This one isn't really required, is it?

anacierdem replied:
It is not required for the clean recipe but for `examples-clean` as the child makefiles does not have the file before anything is installed.

rasky reviewed (commented):
Upstream has reverted the problematic commit. I think you can sync with their trunk if you wish so.

anacierdem commented:
I have also opened [this](https://github.com/buu342/N64-UNFLoader/pull/69).

Regarding Makefile changes:
- The first problem was that the root `install` recipe was only
depending on `libdragon.a` and `libdragonsys.a`, not the complete chain.
This is fixed by making install depend on `libdragon` instead.
- Second problem; example clean recipes using the include are unable to
find it when the parent makefile (`clobber` recipe) is executed. The
root Makefile has access to the build system but children does not until
it is installed. This was also causing `build.sh` to fail.
- I have also changed the sub make commands into `$(MAKE)` such that
make can know they are actually make commands. Also see [this](https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable)
